config/config.go

```diff
@@ -1207,15 +1207,40 @@ func getCPUTargeterConfig(v *viper.Viper) (tracker.TargeterConfig, error) {
 	}
 }
 
-func getDiskSpaceConfig(v *viper.Viper) (requiredAvailableDiskSpace uint64, warningThresholdAvailableDiskSpace uint64, err error) {
-	requiredAvailableDiskSpace = v.GetUint64(SystemTrackerRequiredAvailableDiskSpaceKey)
-	warningThresholdAvailableDiskSpace = v.GetUint64(SystemTrackerWarningThresholdAvailableDiskSpaceKey)
-	switch {
-	case warningThresholdAvailableDiskSpace < requiredAvailableDiskSpace:
-		return 0, 0, fmt.Errorf("%q (%d) < %q (%d)", SystemTrackerWarningThresholdAvailableDiskSpaceKey, warningThresholdAvailableDiskSpace, SystemTrackerRequiredAvailableDiskSpaceKey, requiredAvailableDiskSpace)
-	default:
-		return requiredAvailableDiskSpace, warningThresholdAvailableDiskSpace, nil
-	}
+// getMemoryConfig returns:
+// - requiredAvailableMemory under which the node may shutdown
+// - warningThresholdAvailableMemory under which the node report unhealthy
+func getMemoryConfig(v *viper.Viper) (uint64, uint64, error) {
+	requiredAvailableMemory := v.GetUint64(SystemTrackerRequiredAvailableMemoryKey)
+	warningThresholdAvailableMemory := v.GetUint64(SystemTrackerWarningThresholdAvailableMemoryKey)
+	if warningThresholdAvailableMemory < requiredAvailableMemory {
+		return 0, 0, fmt.Errorf(
+			"%q (%d) < %q (%d)",
+			SystemTrackerWarningThresholdAvailableMemoryKey,
+			warningThresholdAvailableMemory,
+			SystemTrackerRequiredAvailableMemoryKey,
+			requiredAvailableMemory,
+		)
+	}
+	return requiredAvailableMemory, warningThresholdAvailableMemory, nil
+}
+
+// getDiskSpaceConfig returns:
+// - requiredAvailableDiskSpace under which the node may shutdown
+// - warningThresholdAvailableDiskSpace under which the node report unhealthy
+func getDiskSpaceConfig(v *viper.Viper) (uint64, uint64, error) {
+	requiredAvailableDiskSpace := v.GetUint64(SystemTrackerRequiredAvailableDiskSpaceKey)
```

Comment:
nit: I guess this inner logic could be a helper function where the config keys are args